tests/unit/drm_plugin_test.sh

```diff
@@ -225,16 +233,99 @@ function test_gui_control_remote()
   compare_command_sequence '' "$LINENO" 'expected_cmd_seq' "$output"
 }
 
+function test_gui_control_after_reboot_local()
+{
+  local gui_on_after_reboot_cmd='systemctl set-default graphical.target'
+  local gui_off_after_reboot_cmd='systemctl set-default multi-user.target'
+  local bind_cmd='for i in /sys/class/vtconsole/*/bind; do printf "%s\n" 1 > $i; done; sleep 0.5'
+  local unbind_cmd='for i in /sys/class/vtconsole/*/bind; do printf "%s\n" 0 > $i; done; sleep 0.5'
+  local output
+
+  # Remote
+  remote_parameters['REMOTE_IP']='127.0.0.1'
+  remote_parameters['REMOTE_PORT']='8888'
+  remote_parameters['REMOTE_USER']='juca'
+
+  tearDown
+  ssh_part="ssh -p 8888 juca@127.0.0.1"
+  full_turn_on_gui_after_reboot_cmd="${ssh_part} sudo \"${gui_on_after_reboot_cmd}\""
+  full_bind_cmd="${ssh_part} 'sudo bash -c '\''${bind_cmd}'\'"
+
+  declare -a expected_cmd_seq=(
+    "$full_turn_on_gui_after_reboot_cmd"
+    "$full_bind_cmd"
+  )
+
+  output=$(gui_control 'ON_AFTER_REBOOT' '3' '127.0.0.1:8888' 'TEST_MODE')
+  compare_command_sequence '' "$LINENO" 'expected_cmd_seq' "$output"
+
+  full_turn_off_gui_after_reboot_cmd="${ssh_part} sudo \"${gui_off_after_reboot_cmd}\""
+  full_unbind_cmd="$ssh_part 'sudo bash -c '\''${unbind_cmd}'\'"
+
+  declare -a expected_cmd_seq=(
+    "$full_turn_off_gui_after_reboot_cmd"
+    "$full_unbind_cmd"
+  )
```

Comment:
It would be better if each option had its own test; i.e., you could create one function for on and another one for off (same idea but for local and remote as well).